test/test_utils/__init__.py

```diff
@@ -1210,6 +1210,17 @@ def get_cuda_version_from_tag(image_uri):
     if all(keyword in image_uri for keyword in cuda_str):
         cuda_framework_version = re.search(r"(cu\d+)-", image_uri).groups()[0]
 
+    if not cuda_framework_version and "gpu" in image_uri:
```

Comment:
Why not just get the CUDA version from the other tags that **will** be on the image?

Do we know for sure that `CUDA_VERSION` is one of the env variables created in GPU images?